lib/hammer_cli_katello/module_stream.rb

```diff
@@ -0,0 +1,55 @@
+module HammerCLIKatello
+  class ModuleStreamCommand < HammerCLIKatello::Command
+    resource :module_streams
+
+    class ListCommand < HammerCLIKatello::ListCommand
+      output do
+        field :id, _("ID")
+        field :name, _("Module Stream Name")
+        field :stream, _("Stream")
+        field :uuid, _("UUID")
+        field :version, _("Version")
+        field :arch, _("Architecture")
+        field :context, _("Context")
+      end
+
+      build_options
```

Comment:
```sh
$ hammer module-stream list --repository modular --product-id 253
Error: Unrecognised option '--product-id'.
$ hammer module-stream list --repository modular
Error: Found more than one repository.
```

Note, this works fine if you don't have two products with the same repo name, but resolving repo name should always require product ID.

Reply:
would this have to happen on the API side of things? Currently we don't take product id for module streams info endpoint

![here](http://www.clipular.com/c/4879502698872832.png?k=MEC5GUshbc_hBk7je_XdgXOsXeg)

Reply:
No, this should be able to be enabled with the option builder. Something like:

```rb
build_options do |o|
  o.expand.including(:repositories)
  o.expand.including(:products) # or maybe this, I'm not sure
end
```

Reply:
That works for adding the option, but I don't think that fixes the original issue of having two repos in different products with the same name as it doesn't require the `--product-id` params when using `--repository`. 

This also creates (imo) some confusion for the product arguments. I'm seeing this in other repository content types as well so it may be a separate issue to address if I'm not misunderstanding something:

You can't use product-id by itself, but it is a specified option in the help menu. Maybe you can use it for a product with one repository, but that doesn't seem too useful.
```
[vagrant@hammer-devel2 ~]$ hammer package list --product-id 1
Error: Found more than one repository.
[vagrant@hammer-devel2 ~]$ hammer erratum list --product-id 1
Error: Found more than one repository.
``` 
You can query the repo with or without the product id, nothing is enforced.
```
[vagrant@hammer-devel2 ~]$ hammer erratum list --product-id 1 --repository zoo_with_modules
---|----------------|-------------|----------------------
ID | ERRATA ID      | TYPE        | TITLE                
---|----------------|-------------|----------------------
1  | RHEA-2012:0059 | enhancement | Duck_Kangaroo_Erratum
5  | RHEA-2012:0058 | enhancement | Gorilla_Erratum      
4  | RHEA-2012:0056 | security    | Bird_Erratum         
3  | RHEA-2012:0057 | security    | Bear_ErratumPARTHA   
2  | RHEA-2012:0055 | security    | Sea_Erratum          
---|----------------|-------------|----------------------
[vagrant@hammer-devel2 ~]$ hammer erratum list --repository zoo_with_modules
---|----------------|-------------|----------------------
ID | ERRATA ID      | TYPE        | TITLE                
---|----------------|-------------|----------------------
1  | RHEA-2012:0059 | enhancement | Duck_Kangaroo_Erratum
5  | RHEA-2012:0058 | enhancement | Gorilla_Erratum      
4  | RHEA-2012:0056 | security    | Bird_Erratum         
3  | RHEA-2012:0057 | security    | Bear_ErratumPARTHA   
2  | RHEA-2012:0055 | security    | Sea_Erratum          
---|----------------|-------------|----------------------
```

Maybe I'm misunderstanding how this is supposed to function, but product-id doesn't seem to be required when adding it to the build options. I'll keep it there for now since that seems to be the pattern for repository content classes, and will give the option to specify a product when there are repos of the same name, but it doesn't seem enforced.

Reply:
Ah, yes. The second piece is to add option validation. :) Sorry I didn't mention that before. [Here's an example](https://github.com/Katello/hammer-cli-katello/blob/49b48b3c1d431da10c254e3c59e62552cb909122/lib/hammer_cli_katello/content_view.rb#L112). Does that address your concerns?

Reply:
@akofink thanks, that does help. I'll add something like that. I'll probably abstract it out and include it in the other content type classes (errata, package group, etc...) as well since its applicable there.